asdf/extension/_serialization_context.py

```diff
@@ -1,5 +1,8 @@
+import enum
+
+from asdf._block.key import Key as BlockKey
 from asdf._helpers import validate_version
-from asdf.extension import ExtensionProxy
+from asdf.extension._extension import ExtensionProxy
 
 
 class SerializationContext:
```

Comment:
I think that's a great idea. I want to say that's enough of a change that it could go in a separate PR so I'll open an issue to track it: https://github.com/asdf-format/asdf/issues/1626